specs/binding-kafka.spec/src/main/resources/META-INF/zilla/kafka.idl

```diff
@@ -538,24 +538,13 @@ scope kafka
     {
         enum KafkaEventType (uint8)
         {
-            AUTHORIZATION (1),
+            AUTHORIZATION_FAILURE (1),
             API_VERSION_REJECTED (2)
         }
 
-        enum Result (uint8)
-        {
-            SUCCESS (0),
-            FAILURE (1)
-        }
-
-        struct KafkaAuthorizationEvent extends core::event::Event
-        {
-            Result result;
-        }
-
         union KafkaEvent switch (KafkaEventType)
         {
-            case AUTHORIZATION: KafkaAuthorizationEvent authorization;
+            case AUTHORIZATION_FAILURE: core::event::Event authorizationFailure;
```

Comment:
```suggestion
            case AUTHORIZATION_FAILED: KafkaAuthorizationFailed authorizationFailed;
            case API_VERSION_REJECTED: KafkaApiVersionRejected apiVersionRejected;
```

We need separate event structures to carry relevant information. We will need to sort out how to get attempted authorization identity from the guard, so authorization failed events will need identity, and api version rejected needs to know which api and which version.